samples/sample_vpp/src/sample_vpp.cpp

```diff
@@ -731,58 +731,62 @@ int main(int argc, msdk_char *argv[])
             else
             {
 #ifdef ENABLE_VPP_RUNTIME_HSBC
-                auto procAmp = pOutSurf->AddExtBuffer<mfxExtVPPProcAmp>();
-                // set default values for ProcAmp filters
-                procAmp->Brightness = 0.0F;
-                procAmp->Contrast   = 1.0F;
-                procAmp->Hue        = 0.0F;
-                procAmp->Saturation = 1.0F;
-
-                if (Params.rtHue.isEnabled)
+                if (Params.rtHue.isEnabled || Params.rtSaturation.isEnabled ||
+                    Params.rtBrightness.isEnabled || Params.rtContrast.isEnabled)
                 {
-                    if((nOutFrames / Params.rtHue.interval & 0x1) == 0)
+                    auto procAmp = pOutSurf->AddExtBuffer<mfxExtVPPProcAmp>();
+                    // set default values for ProcAmp filters
+                    procAmp->Brightness = 0.0F;
+                    procAmp->Contrast   = 1.0F;
+                    procAmp->Hue        = 0.0F;
+                    procAmp->Saturation = 1.0F;
+
+                    if (Params.rtHue.isEnabled)
                     {
-                        procAmp->Hue = Params.rtHue.value1;
+                        if((nOutFrames / Params.rtHue.interval & 0x1) == 0)
+                        {
+                            procAmp->Hue = Params.rtHue.value1;
```

Comment:
@MicroYY , generally the change is right. But since you're touching this code :) , can you please re-factor it a bit , this way:
`procAmp->Hue = ((nOutFrames / Params.rtHue.interval & 0x1) == 0) ?  Params.rtHue.value1 : Params.rtHue.value2`
?
and the same for other procAmp settings.